### PR TITLE
Static comission from validators to nominators for staking

### PR DESCRIPTION
**Pull request by akorchyn**
- [x] Implement median or static commission to nominators from validator preferences
- [x] Configured by referenda (OpenGov). 
- [x] Add tests 

Notes:
* The commission median is calculated from the actual validator preferences, not the ERA-ones. So, the validator can apply changes on the fly, which will be used in calculating the next session payout.

### CODE REVIEW

**Review by baidang201 — APPROVED** [review]

File: runtime/mainnet/src/pos/session_payout/mod.rs [hunk 1]
```diff
@@ -18,6 +21,37 @@ type PositiveImbalanceOf<T> =
...
+				let current_validators = pallet_staking::ErasStakers::<T>::iter_key_prefix(era);
+				let mut comissions = current_validators
+					.map(|id| pallet_staking::Validators::<T>::get(id).commission)
+					.collect::<Vec<_>>();
+
+				if comissions.is_empty() {
+					return None;
+				}
+				comissions.sort_unstable();
+				let median = comissions[comissions.len() / 2];
```
Comment (obsessed-cake) [inline runtime/mainnet/src/pos/session_payout/mod.rs:48]:
```suggestion
				let median = if comissions.len() % 2 == 1 {
				  comissions[comissions.len() / 2]
				} else {
				  comissions[comissions.len() / 2 - 1].saturating_add(comissions[comissions.len() / 2 + 1]) / 2
				}
```

Comment (obsessed-cake) [reply runtime/mainnet/src/pos/session_payout/mod.rs:48]: See [wiki](https://en.wikipedia.org/wiki/Median) for details.

File: runtime/mainnet/src/pos/session_payout/mod.rs [hunk 2]
```diff
@@ -304,67 +365,64 @@ where
...
-		// Make payout at the end of each session.
-		let treasury_commission = T::CurrencyInfo::treasury_commission_from_staking();
-
-		let now_as_millis = T::TimeProvider::now().as_millis();
-		let last_payout = SessionStartTime::<T>::get();
-		let session_duration = now_as_millis - last_payout;
-
-		Self::update_year_reward(now_as_millis);
+		} else {
+			let current_era = current_era.unwrap();
```
Comment (obsessed-cake) [inline runtime/mainnet/src/pos/session_payout/mod.rs:386]: we can avoid `unwrap` so:
```
if let Some(current_era) = current_era {
  // ...
} else {
  log::warn!(target: "runtime::session_payout", "end_session: current_era is None");
}

File: runtime/mainnet/src/pos/session_payout/mod.rs (hunk 1 above)
Comment (akorchyn) [reply runtime/mainnet/src/pos/session_payout/mod.rs]: Done

File: runtime/mainnet/src/pos/session_payout/mod.rs (hunk 2 above)
Comment (akorchyn) [reply runtime/mainnet/src/pos/session_payout/mod.rs]: Done

**Review by obsessed-cake — APPROVED** [review]

**Pull request merged**